Mathlib/RingTheory/NonZeroDivisors.lean

```diff
@@ -178,3 +193,20 @@ theorem prod_zero_iff_exists_zero [NoZeroDivisors M₁] [Nontrivial M₁] {s : M
 #align prod_zero_iff_exists_zero prod_zero_iff_exists_zero
 
 end nonZeroDivisors
+
+section nonZeroSMulDivisors
+
+open nonZeroSMulDivisors nonZeroDivisors
+
+variable (R M : Type*) [MonoidWithZero R] [Zero M] [MulAction R M]
+
+lemma mem_nonZeroSMulDivisors_iff (x : R) : x ∈ R⁰[M] ↔ ∀ (m : M), x • m = 0 → m = 0 := Iff.rfl
+
+lemma nonZeroSMulDivisors_over_self_eq_nonZeroDivisors_of_mul_comm
+    (comm : ∀ (r s : R), r * s = s * r) :
+    R⁰[R] = R⁰ := by
+  ext r : 1
+  rw [mem_nonZeroSMulDivisors_iff R R, mem_nonZeroDivisors_iff]
+  simp only [smul_eq_mul, comm]
```

Comment:
when you say reuse `nonZeroDivisors`, do you mean something like this @eric-wieser

Reply:
No, I meant `(MulOpposite R)⁰[R] = R⁰` which should be true by `rfl` without `comm`

Reply:
This has bad typing `(MulOpposite R)⁰[R]` has type `Submonoid Rᵐᵒᵖ`

Reply:
`R⁰[MulOpposite R] = R⁰` I can't get `rfl` on this either

Reply:
Ah, I think I mean `Submonoid.unop (MulOpposite R)⁰[R]` if that exists

Reply:
I come up with something like this
```lean

def unop' (x : Submonoid (MulOpposite R)) : Submonoid R where
  carrier := MulOpposite.unop '' x
  mul_mem' := sorry
  one_mem' := sorry

def op' (x : Submonoid R) : Submonoid (MulOpposite R) where
  carrier := MulOpposite.op '' x
  mul_mem' := sorry
  one_mem' := sorry

example : (@unop' R _ ((MulOpposite R)⁰[R]) : Submonoid R) = R⁰ := sorry
example : (@op' R _ R⁰) = (MulOpposite R)⁰[R] := sorry
```
I still don't think the proofs are `rfl` though

Reply:
They should be defined with the preimage not the image